basement/Basement/Bits.hs

```diff
@@ -291,9 +291,15 @@ instance FiniteBitsOps Word where
     rotateL w (CountOf i) = w `OldBits.rotateL` i
     rotateR w (CountOf i) = w `OldBits.rotateR` i
     bitFlip = OldBits.complement
+#if __GLASGOW_HASKELL__ >= 904
+    popCount (W# x#) = CountOf $ wordToInt (W# (popCnt64# (wordToWord64# x#)))
+    countLeadingZeros (W# w#) = CountOf $ wordToInt (W# (clz64# (wordToWord64# w#)))
+    countTrailingZeros (W# w#) = CountOf $ wordToInt (W# (ctz64# (wordToWord64# w#)))
+#else
```

Comment:
Relevant note in migration guide: https://gitlab.haskell.org/ghc/ghc/-/wikis/migration/9.4#word64int64-use-word64int64